addons/sale/models/sale_order.py

```diff
@@ -1407,12 +1422,15 @@ def show_line(line):
 
     def _get_default_payment_link_values(self):
         self.ensure_one()
+        amount = self.amount_total - self.amount_paid
         return {
             'description': self.name,
-            'amount': self.amount_total - sum(self.invoice_ids.filtered(lambda x: x.state != 'cancel' and x.invoice_line_ids.sale_line_ids.order_id == self).mapped('amount_total')),
             'currency_id': self.currency_id.id,
             'partner_id': self.partner_id.id,
-            'amount_max': self.amount_total,
+            'amount': amount,
+            'amount_max': amount,
```

Comment:
Both the same?

Reply:
Yep, do not allow an amount bigger that what remains to be paid and let the initial amount to be equal to the maximal amount that can be paid